Automattic/wp-calypso
# i18n: Use bundeled momentjs data instead of our own

In the current i18n process, we overwrite the bundled momentjs [locale data](https://github.com/moment/moment/tree/develop/locale), with our own data generated  in our exports from translate.wordpress.com. While this has the benefit of some flexibility around fixing localization issues, it also comes with some major disadvantages:
- Duplicate translation work: momentjs already has this translated in many locales. 
- Loss of advanced functionality, like the ability to define a translation via a function ([example](https://github.com/moment/moment/blob/develop/locale/he.js#L69))
- Developers expect the translations to match the examples from momentjs (see #2642). Diverging from that creates confusion.
- Bigger locale files

This PR switches to using the default bundled data, and removes the custom strings from our translations process (Calypso side):

**Considerations:** 
- We now add the locales file to the webpack bundle. I'm not sure how to verify the difference in file size. All locale files take about 0.5MB on disk.
- If we find ourselves in a situation where the momentjs locale data is incomplete or wrong, we can overwrite in code, while pushing fixes uprstream.

**Testing**
- Switch to translated non english locale (i.e 'es' or 'fr')
- Check that localized dates are still localized. For example, the date on hover over a post human date in the reader.
